### ISSUE TITLE
LC_RPATH was gone and thus vulkan supported mpv won't start...

### ISSUE THREAD

**Issue by xanadu6291** [mpv-player/mpv#13628]
### Important Information

Provide following Information:
- mpv version
  0.37.0-476-g9ac791c329

- macOS Version
  14.3.1

- Source of the mpv binary or bundle
  git hub

- If known which version of mpv introduced the problem
 Not sure

- Possible screenshot or video of visual glitches
Since it's a start up issue, there's no screen shot

If you're not using git master or the latest release, update.
Releases are listed here: https://github.com/mpv-player/mpv/releases

### Reproduction steps

just run the command `mpv ~/Desktop/video.mp4`

Try to reproduce your issue with --no-config first. If it isn't reproducible
with --no-config try to first find out which option or script causes your issue.

Describe the reproduction steps as precise as possible. It's very likely that
the bug you experience wasn't reproduced by the developer because the workflow
differs from your own.

### Expected behavior

mpv plays video correctly.

### Actual behavior

./mpv ~/Desktop/video.mp4 
dyld[13569]: Library not loaded: @rpath/libshaderc_shared.1.dylib
  Referenced from: <784E33B7-9C21-3F41-B3BB-7E904A912210> /Volumes/mpv_compile/bin/mpv
  Reason: no LC_RPATH's found
Abort trap: 6

After investigation, I found the following:
I am using disk image named mpv_compile, and installing related files into it.
Executable file mpv which is placed in /Volumes/mpv_compile/mpv/build works fine.
The above have LC_RPATH. I checked it with `otool -l | grep -A2 LC_RPATH`
Executable file mpv which is placed in /Volumes/mpv_compile/bin, which is created by the command `meson install -C build` does not work.
The above does not have LC_RPATH at all. Very funny...

### Log file

No log file.

Make a log file made with -v -v or --log-file=output.txt, attach it to
the issue, and replace this text with a link to it.

If you use the Bundle, a default log is created for your last run at
~/Library/Logs/mpv.log. You can jump to that file via the
Help > Show log File… menu.

Without the log file, this issue will be closed for ignoring the issue template.

In the case of a crash, please provide the macOS Crash Report (Backtrace).

### Sample files

Sample files needed to reproduce this issue can be attached to the issue
(preferred), or be uploaded to https://0x0.st/ or similar sites.
(Only needed if the issue cannot be reproduced without it.)
Do not use garbage like "cloud storage", especially not Google Drive.

Since it's a start up issue, there's no sample files

**Comment by Akemi** [comment]
possibly a meson problem? 
https://github.com/mesonbuild/meson/issues/12045
https://github.com/mesonbuild/meson/issues/11109
https://github.com/mesonbuild/meson/issues/10711

**Comment by Akemi** [comment]
i can reproduce the exact same problem. maybe this specific problem is related to https://github.com/mpv-player/mpv/commit/60a31a8a284e539043c47a33ff90f3d55aade6a8 it at least worked at some point.

<img width="1772" alt="Screenshot 2024-03-04 at 19 55 23" src="https://github.com/mpv-player/mpv/assets/680386/75b06809-69e7-4675-bea4-0f33b257d38b">

[build.txt](https://github.com/mpv-player/mpv/files/14485978/build.txt)
[install.txt](https://github.com/mpv-player/mpv/files/14485979/install.txt)

**Comment by xanadu6291** [comment]
> possibly a meson problem? [mesonbuild/meson#12045](https://github.com/mesonbuild/meson/issues/12045) [mesonbuild/meson#11109](https://github.com/mesonbuild/meson/issues/11109) [mesonbuild/meson#10711](https://github.com/mesonbuild/meson/issues/10711)

Yes, the three of them seems to similar with my problem. Should I have to contact with meson development team?

**Comment by xanadu6291** [comment]
> i can reproduce the exact same problem. ~maybe this specific problem is related to [60a31a8](https://github.com/mpv-player/mpv/commit/60a31a8a284e539043c47a33ff90f3d55aade6a8).~ it at least worked at some point.
> 
> <img alt="Screenshot 2024-03-04 at 19 55 23" width="1772" src="https://private-user-images.githubusercontent.com/680386/309869286-75b06809-69e7-4675-bea4-0f33b257d38b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk2MTA4OTEsIm5iZiI6MTcwOTYxMDU5MSwicGF0aCI6Ii82ODAzODYvMzA5ODY5Mjg2LTc1YjA2ODA5LTY5ZTctNDY3NS1iZWE0LTBmMzNiMjU3ZDM4Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwNVQwMzQ5NTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01OTEyY2Q0ZDNjMzEzNThhZjMyMzM1ZmFiYjRiYjdhNGU1NTY3NTAxMzdiYzE1OTcwNDIyYjcxMGQ5ZGFjZDVlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.y6VD_ZMskb7i0Qnbsi-X540pEBznxHteythrvedggLA">
> [build.txt](https://github.com/mpv-player/mpv/files/14485978/build.txt) [install.txt](https://github.com/mpv-player/mpv/files/14485979/install.txt)

Result of `diff -u build.txt install.txt` is exactly same as my problem. By the way, I forgotten to write messon version. That is 1.3.2 and ninja version is 1.11.1

**Comment by Akemi** [comment]
you can probably open a new issue on the meson issue tracker, though since it has probably already been reported it might be closed.

**Comment by Akemi** [comment]
seems like it was fixed 3 days ago 
https://github.com/mesonbuild/meson/pull/13012
https://github.com/grobian/meson/commit/78e9009ff9d36925e04f329f9082841002ddd848

for now we need to wait for a new release, it will probably be in 1.4.1.

**Comment by xanadu6291** [comment]
> seems like it was fixed 3 days ago [mesonbuild/meson#13012](https://github.com/mesonbuild/meson/pull/13012) [mesonbuild/meson@78e9009](https://github.com/mesonbuild/meson/commit/78e9009ff9d36925e04f329f9082841002ddd848)
> 
> for now we need to wait for a new release, it will probably be in 1.4.1.

That's fine to hear. I will wait for next release of meson (1.4.1). Thanks for information...

**Comment by Akemi** [comment]
meson 1.4.1 has been released and on my end the issue has been fixed. if there are still problems feel free to open a new issue or let me know.

**Comment by xanadu6291** [comment]
I've got a similar issue with meson version 1.5.1.  This time, two of LC_RPATH exists, but LC_RPATH of /usr/local/lib and LC_RPATH of /usr/local/vulkan/lib have gone...  Rollbacking meson to version 1.4.1, the issue have gone...

**Comment by xanadu6291** [comment]
I've got a similar issue with meson version 1.5.1.  This time, two of LC_RPATH exists, but LC_RPATH of /usr/local/lib and LC_RPATH of /usr/local/vulkan/lib have gone...  I think these two LC_RPATH should be exists because they exists 
 at the build directory of mpv.  Rollbacking meson to version 1.4.1, the issue have gone...

**Comment by Akemi** [comment]
https://github.com/mesonbuild/meson/commit/7b43a2e19613f3702935872b21b1020f7d8c6a9b

seems like only this commit was added to the previous related code. you probably need to report it to meson.

**Comment by xanadu6291** [comment]
I have filed new issue at meson project: 

[mpv won't start with meson install -C build command #13534](https://github.com/mesonbuild/meson/issues/13534)

Please correct if needed...
